models/repo_list.go

```diff
@@ -136,6 +151,16 @@ const (
 // SearchRepositoryByName takes keyword and part of repository name to search,
 // it returns results in given range and number of total results.
 func SearchRepositoryByName(opts *SearchRepoOptions) (RepositoryList, int64, error) {
+	var onlyOwnersRepo = opts.SearchMode == SearchModeFork || opts.SearchMode == SearchModeSource
+
+	if onlyOwnersRepo && opts.OwnerID <= 0 {
+		return nil, 0, nil
```

Comment:
Shouldn't this result in error if search params are wrong (from UI there should not be such way to get there and from api you need to know that you are calling it wrong)

Reply:
That's question... Params are not wrong, only combination of some parameteres can result in empty result. As I know what combinations will return empty array of repositories I included these 2 tests at start to avoid unnecesary and slow db search.